pkg/autodiscovery/configmgr.go

```diff
@@ -78,216 +78,357 @@ type configManager interface {
 	mapOverLoadedConfigs(func(map[string]integration.Config))
 }
 
-// simpleConfigManager implements the "simple" config manager that reconciles
-// services and templates without any priority, using a store as a backend.
-type simpleConfigManager struct {
+// serviceAndADIDs bundles a service and its associated AD identifiers.
+type serviceAndADIDs struct {
+	svc   listeners.Service
+	adIDs []string
+}
+
+// a multimap is a string/string map that can contain multiple values for a
+// single key.  Duplicate values are allowed (but not used in this package).
+type multimap struct {
```

Comment:
NIT: Maybe `multimap` can go in it's own file? I feel like it's on the edge of being trivial enough to include inline - but this file is already rather large.